packages/venia-concept/src/components/SearchBar/autocomplete.js

```diff
@@ -65,6 +65,17 @@ class SearchAutocomplete extends Component {
 
     handleOnProductOpen = () => this.props.updateAutocompleteVisible(false);
 
+    // map Magento 2.3.1 schema changes to v2.0.0 proptype shape to maintain backwards compatibility
+    mapProducts(products) {
+        return products.map(product => {
+            if (typeof product.small_image === 'object') {
+                product.small_image = product.small_image.url;
+            }
```

Comment:
Echoing @sirugh 's sentiments earlier, this does seem like the same transform each time so it could be nice to re-use this logic.

Understandable if that's a future nice to have.

Reply:
@jimbo can elaborate more, but I think the intention is that this code would be obsolete in the next major, and be removed. Given it's temporary, so we didn't want to over engineer a solution that would just be deleted in the near future.

Reply:
That's exactly the idea. It's the same idea as our requirement that all REST calls specify the URL and stringify the payload inline. It's a way of emphasizing that the REST calls are temporary workarounds, and making it more obvious how big a change it is when they are removed.

Reply:
Immutable version:

```js
mapProducts(products) {
  return products.map(product => {
    const { small_image } = product;
    return {
      ...product,
      small_image: typeof small_image === 'object' ? small_image.url : small_image 
    }
  })
}
```